scheduler/filter/affinity.go

```diff
@@ -34,7 +34,10 @@ func (f *AffinityFilter) Filter(config *cluster.ContainerConfig, nodes []*node.N
 			case "container":
 				containers := []string{}
 				for _, container := range node.Containers {
-					containers = append(containers, container.Id, strings.TrimPrefix(container.Names[0], "/"))
+					containers = append(containers, container.Id)
+					if len(container.Names) > 0 {
+						containers = append(containers, strings.TrimPrefix(container.Names[0], "/"))
```

Comment:
@vieux is it possible for the length of `container.Names` is 0 ? (which means the container doesn't have a name)